packages/data-context/src/sources/VersionsDataSource.ts

```diff
@@ -11,53 +19,116 @@ interface VersionData {
   latest: Version
 }
 
+const REMOTE_MANIFEST_URL = 'https://download.cypress.io/desktop.json'
+const NPM_CYPRESS_REGISTRY = 'https://registry.npmjs.org/cypress'
+
 export class VersionsDataSource {
-  static result: undefined | Promise<string>
+  private _initialLaunch: boolean
+  private _currentTestingType: TestingType | null
+  private _latestVersion: Promise<string>
+  private _npmMetadata: Promise<Record<string, string>>
+
+  constructor (private ctx: DataContext) {
+    this._initialLaunch = true
+    this._currentTestingType = this.ctx.coreData.currentTestingType
+    this._latestVersion = this.getLatestVersion()
+    this._npmMetadata = this.getVersionMetadata()
+  }
 
   /**
    * Returns most recent and current version of Cypress
    * {
    *   current: {
+   *     id: '8.7.0',
    *     version: '8.7.0',
    *     released: '2021-10-15T21:38:59.983Z'
    *   },
    *   latest: {
+   *     id: '8.8.0',
    *     version: '8.8.0',
    *     released: '2021-10-25T21:38:59.983Z'
    *   }
    * }
    */
-  async versions (): Promise<VersionData> {
-    const currentCypressVersion = require('@packages/root')
+  async versionData (): Promise<VersionData> {
+    this.resetLatestVersionTelemetry()
```

Comment:
If we keep call this here, then we're potentially going to be calling `getLatestVersion` earlier than we want, right?

Reply:
I'm not sure what you mean. Once we're in this function we'd always want to call it right?

Reply:
[`f81145c` (#20107)](https://github.com/cypress-io/cypress/pull/20107/commits/f81145c0d9c91e4f26f1d16cfce5e5997053db21)